### PR TITLE
Make search_form twig component independant

### LINKED ISSUE #33177: Independent twig component: Search form
## Description

See #33101 for description in Step 2

This is the Search form part.

This component should be quite autonomous.

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Make search_form twig component callable independantly
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI and auto tests green
| Fixed ticket?     | Fixes #33177
| Related PRs       | 
| Sponsor company   | PrestaShop SA

https://github.com/matthieu-rolland/ga.tests.ui.pr/actions/runs/5690528844

### CODE REVIEW

**Review by mflasquin — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

File: src/PrestaShopBundle/Twig/Component/SearchForm.php [hunk 1]
```diff
@@ -28,20 +28,28 @@
...
+    {
+    }
 
-    public function mount(bool $showClearBtn = null, bool $searchType = null): void
+    public function mount(): void
     {
-        $this->showClearBtn = $showClearBtn ?? false;
-        $this->searchType = $searchType ?? false;
+        $request = $this->requestStack->getCurrentRequest();
+        $this->boQuery = $request->query->get(self::BO_QUERY_PARAM) ?: '';
```
Comment (lartist) [inline src/PrestaShopBundle/Twig/Component/SearchForm.php]:
```suggestion
        $this->boQuery = $request->query->get(self::BO_QUERY_PARAM, '');
```

File: src/PrestaShopBundle/Twig/Component/SearchForm.php [hunk 2]
```diff
@@ -28,20 +28,28 @@
...
+        $this->searchType = (int) $request->query->get(self::BO_SEARCH_TYPE_PARAM) ?: 0;
```
Comment (lartist) [inline src/PrestaShopBundle/Twig/Component/SearchForm.php]:
```suggestion
        $this->searchType = (int) $request->query->getInt(self::BO_SEARCH_TYPE_PARAM, 0);
```

**Review by lartist — CHANGES_REQUESTED** [review]
Please remove the following lines of phpstan-tmp-legacy-layout-baseline.neon
```neon
		-
			message: "#^Class Link is forbidden, No legacy calls inside the prestashop bundle\\. Please create an interface and an adapter if you need to\\.$#"
			count: 1
			path: src/PrestaShopBundle/Twig/Component/SearchForm.php

		-
			message: "#^Namespace Link is forbidden, No legacy calls inside the prestashop bundle\\. Please create an interface and an adapter if you need to\\.$#"
			count: 1
			path: src/PrestaShopBundle/Twig/Component/SearchForm.php
```

And also in admin-dev/themes/new-theme/template/layout.tpl please remove unused variables:
              {render_template
                smarty_template="components/layout/search_form.tpl"
                twig_template="@PrestaShop/Admin/Layout/search_form.html.twig"
                ~~bo_query=$bo_query
                link=$link
                show_clear_btn= $show_clear_btn|default:null
                search_type= $search_type|default:null~~
              }

File: src/PrestaShopBundle/Resources/views/Admin/Layout/search_form.html.twig [hunk 3]
```diff
@@ -22,9 +22,4 @@
  * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  *#}
-{{ component('SearchForm', {
-  boQuery: bo_query,
-  link: link,
-  showClearBtn: show_clear_btn,
-  searchType: search_type
-}) }}
```
Comment (M0rgan01) [inline src/PrestaShopBundle/Resources/views/Admin/Layout/search_form.html.twig:30]: You can also remove them in layout.tpl

**Review by boherm — DISMISSED** [review]

**Comment by lartist** [comment]
@matthieu-rolland I think you forget this in your last commit:
> And also in admin-dev/themes/new-theme/template/layout.tpl please remove unused variables:
{render_template
smarty_template="components/layout/search_form.tpl"
twig_template="@PrestaShop/Admin/Layout/search_form.html.twig"
bo_query=$bo_query
link=$link
show_clear_btn= $show_clear_btn|default:null
search_type= $search_type|default:null
}

**Review by boherm — DISMISSED** [review]

**Review by M0rgan01 — DISMISSED** [review]

**Review by lartist — DISMISSED** [review]

**Review by aniszr — DISMISSED** [review]
Hello @matthieu-rolland 

I have checked your PR and I noticed this issue :x: 

- After activating the Symfony layout option, the access to some pages became impossible  :x: 

1. Orders > Orders : See attached screen-record:

https://github.com/PrestaShop/PrestaShop/assets/92626242/9c25316d-35a4-4146-bd11-5dc4af62ba7c

2.Modules > Modules  Manager

image

3. Catalog > Categories
4. Catalog > Brands & Suppliers
5. Catalog > Files

6. Customers > Customers
7. Customers > Addresses

8. Customer Service > Order Messages

9.Design > Theme & Logo
10.Email Theme
11. Design > Pages  
image

12.Design > Positions
13.Design > Link List

14. International > Localization
15. International > Locations
16. International > Taxes

17. Shop Parameters > General
18. Shop Parameters >Order Setings
19. Shop Parameters > Customer Settings
20. Shop Parameters >Contact
21. Shop Parameters >Traffic & SEO

22. Advanced Parameters > Performance

image

23.Advanced Parameters > Team
24.Advanced Parameters > Database
25.Advanced Parameters > Webservice
26.Advanced Parameters > Security

Auto Test :heavy_check_mark:  : https://github.com/aniszr/ga.tests.ui.pr/actions/runs/5748987216

Could you please check and feed-back!

Thanks!

**Comment by matthieu-rolland** [comment]
@aniszr I have been told that this PR should not be tested by QA because it's behind a feature flag, I didn't know sorry about that 😓 

About your findings, I couldn't reproduce them, it's strange 🤔 

**EDIT**: I just reproduced it, not sure why I didn't have them at first, maybe some cache, I'll look into it

**EDIT 2:** Seems like this issue is also on develop branch.

**Comment by matthieu-rolland** [comment]
> QA OK without required approvals !? :trollface:

whatcha gonna do ?

**Review by boherm — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Pull request merged**